### PR TITLE
ACI push pool guide

### PR DESCRIPTION
**Pull request by zhen0**
Adds docs for adding an azure container instance push work pool.  I'm adding notes based on my experience setting up azure and would welcome a review/input from @prefectcboyd who I consider to be much more of an azure expert. It's likely that I go into a bit much detail on navigating the azure portal/UI. 

I run into an issue when serving the docs locally because my version of pymdown extensions doesn't like our addition of `pymdownx.tabbed`.  Taking out `pyndownx.tabbed`  from our config lets me serve the docs but the tabs we're using for the push work pools guide don't work. Equally, it seems our netlify build isn't working either.  So I'm leaving this as draft until we can properly check the deploy preview and make sure tabs are correctly displayed. 

### Example

 
 Example - but the tabs aren't working as I'd expect:
 
 
image

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

File: docs/guides/push-work-pools.md [hunk 1]
```diff
@@ -14,13 +15,14 @@ Push [work pools](/concepts/work-pools/#work-pool-overview) are a special type o
 
 In this guide you will:
 
-- Create a push work pool that sends work to Google Cloud Run or AWS ECS
+- Create a push work pool that sends work to Google Cloud Run, AWS ECS or Azure
```
Comment (prefectcboyd) [inline docs/guides/push-work-pools.md:18]: Thoughts on Azure ACI - ? It keeps it in line with "Provider -> Offering" for the Google / AWS examples

File: docs/guides/push-work-pools.md [hunk 2]
```diff
@@ -32,12 +34,35 @@ In this guide you will:
...
+    ### Create App Registration
+
+    1. In the Azure portal, create an app registration.
+    2. In the app registration, create a client secret. Copy the value and store it somewhere safe.
+    
+    ### Add App Registration to Subscription
+
+    1. Navigate to the resource group you created earlier.
+    2. Click on "Access control (IAM)" and then "Role assignments".
+    3. Search for the app registration and select it. Give it a role that has sufficient privileges to create, run, and delete ACI container groups.
```
Comment (prefectcboyd) [inline docs/guides/push-work-pools.md:62]: They can either add contributor to a specific resource group (that they created) to limit the scope, or add a custom role / permissions with explicit `Microsoft.ContainerInstance/containerGroups/*` - still probably good to scope that to the resource group. 

I'm not sure how detailed we need to be here though for users - RBAC and permissions should fall to them?

**Review by prefectcboyd — COMMENTED** [review]
Left a couple updates, otherwise looks good!

File: docs/guides/push-work-pools.md (hunk 1 above)
Comment (zhen0) [reply docs/guides/push-work-pools.md:18]: Yeah good call out

File: docs/guides/push-work-pools.md (hunk 2 above)
Comment (zhen0) [reply docs/guides/push-work-pools.md:62]: Yeah being too prescriptive on RBAC and permissions get difficult. I think I'll leave as is and we can update with more info if we get questions about it.

**Comment by zhen0** [comment]
> Left a couple updates, otherwise looks good!

Thank you!

**Review by WillRaphaelson — APPROVED** [review]
thanks jenny!

**Pull request merged**